Makefile

```diff
@@ -1,11 +1,56 @@
-BUILD_DIR:=./bin
+BUILDFLAGS ?=
+unexport GOFLAGS
 
-build: clean
-	mkdir -p ${BUILD_DIR}
-	go build -o "${BUILD_DIR}/backplane-tools" main.go
+BASE_DIR=$(shell pwd)
+BIN_DIR=${BASE_DIR}/bin
 
-clean:
-	rm -rf "${BUILD_DIR}"
+all: format mod build test lint
 
+format: vet mod fmt mockgen ci-build docs
+
+fmt:
+	@echo "gofmt"
+	@gofmt -w -s .
+	@git diff --exit-code .
+
+OS := $(shell go env GOOS | sed 's/[a-z]/\U&/')
+ARCH := $(shell go env GOARCH)
+
+.PHONY: download-goreleaser
+download-goreleaser:
+	GOBIN=${BIN_DIR} go install github.com/goreleaser/goreleaser@v1.15.0
+
+.PHONY: download-golangci-lint
+download-golangci-lint:
+	GOBIN=${BIN_DIR} go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.53.3
```

Comment:
**suggestion:** Move the version of golangci-lint and goreleaser to new variables:

e.g.
```
GOLANGCI_LINT_VERSION=v1.53.3
...
.PHONY: download-golangci-lint
download-golangci-lint:
	GOBIN=${BIN_DIR} go install github.com/golangci/golangci-lint/cmd/golangci-lint@${GOLANGCI_LINT_VERSION}	
```